gammapy/spectrum/tests/test_cosmic_ray.py

```diff
@@ -1,13 +1,59 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
-from astropy.units import Quantity
+import pytest
+import astropy.units as u
 from ...utils.testing import assert_quantity_allclose
-from ...spectrum import cosmic_ray_flux
+from ...spectrum import cosmic_ray_spectrum
 
+Cosmic_rays_SPECTRA = [
+    dict(
+        name="proton",
+        dnde=u.Quantity(0.014773732960138994, "m-2 s-1 TeV-1 sr-1"),
+        flux=u.Quantity(0.056470139673467444, "m-2 s-1 sr-1"),
+        index=2.70,
+    ),
+    dict(
+        name="N",
+        dnde=u.Quantity(0.0115347902543466, "m-2 s-1 TeV-1 sr-1"),
+        flux=u.Quantity(0.043840936324311894, "m-2 s-1 sr-1"),
+        index=2.64,
+    ),
+    dict(
+        name="Si",
+        dnde=u.Quantity(0.0044934359085944935, "m-2 s-1 TeV-1 sr-1"),
+        flux=u.Quantity(0.017108254587645998, "m-2 s-1 sr-1"),
+        index=2.66,
+    ),
+    dict(
+        name="Fe",
+        dnde=u.Quantity(0.0021646909913177995, "m-2 s-1 TeV-1 sr-1"),
+        flux=u.Quantity(0.008220752990527653, "m-2 s-1 sr-1"),
+        index=2.63,
+    ),
+    dict(
+        name="electron",
+        dnde=u.Quantity(8.0665368e-06, "m-2 s-1 TeV-1 sr-1"),
+        flux=u.Quantity(3.733755e-05, "m-2 s-1 sr-1"),
+    ),
+]
 
-def test_cosmic_ray_flux():
-    energy = Quantity(1, "TeV")
-    actual = cosmic_ray_flux(energy, "proton")
-    desired = Quantity(0.096, "m-2 s-1 sr-1 TeV-1")
-    assert_quantity_allclose(actual, desired)
 
-    # TODO: test array quantities and other particles
+@pytest.mark.parametrize("spec", Cosmic_rays_SPECTRA, ids=lambda _: _["name"])
+def test_cosmic_ray_spectrum(spec):
+    energy = 2 * u.TeV
+    emin, emax = [1, 1e3] * u.TeV
+
+    cr_spectrum = cosmic_ray_spectrum(particle=spec["name"])
+
+    dnde = cr_spectrum(energy)
+    assert_quantity_allclose(dnde, spec["dnde"])
+
+    flux = cr_spectrum.integral(emin, emax)
+    assert_quantity_allclose(flux, spec["flux"])
+
+    if spec["name"] != "electron":
```

Comment:
Suggest to put the arbitrary ref value for electron also, and to remove this "if" in the test.